apps/web/src/components/common/EnhancedTable/index.tsx

```diff
@@ -145,29 +149,13 @@ function EnhancedTable({ rows, headCells, mobileVariant }: EnhancedTableProps) {
           <EnhancedTableHead headCells={headCells} order={order} orderBy={orderBy} onRequestSort={handleRequestSort} />
           <TableBody>
             {pagedRows.length > 0 ? (
-              pagedRows.map((row, index) => (
-                <TableRow
-                  data-testid="table-row"
-                  tabIndex={-1}
-                  key={row.key ?? index}
-                  selected={row.selected}
-                  className={row.collapsed ? css.collapsedRow : undefined}
-                >
-                  {Object.entries(row.cells).map(([key, cell]) => (
-                    <TableCell
-                      key={key}
-                      className={classNames({
-                        sticky: cell.sticky,
-                        [css.collapsedCell]: row.collapsed,
-                      })}
-                    >
-                      <Collapse key={index} in={!row.collapsed} enter={false}>
-                        {cell.content}
-                      </Collapse>
-                    </TableCell>
-                  ))}
-                </TableRow>
-              ))
+              pagedRows.map((row, index) =>
+                !!row.safenetBalance ? (
+                  <SafenetAssetsRow key={row.key ?? index} row={row} index={index} />
+                ) : (
+                  <AssetsRow key={row.key ?? index} row={row} index={index} />
```

Comment:
The `EnhancedTable` is not just used for Assets.
I think ideally we find a way to achieve this without having to introduce such domain knowledge into this common component.

Reply:
Maybe we can keep this general an call them "embedded rows" or something like that - unrelated to Safenet.